src/window/building/figures.c

```diff
@@ -181,6 +187,34 @@ static void draw_trader(building_info_context *c, figure *f)
             }
         }
     }
+
+    if (city_buildings_has_caravanserai() && building_monument_working(BUILDING_CARAVANSERAI) && f->type != FIGURE_TRADE_SHIP) {
```

Comment:
If `building_monument_working(BUILDING_CARAVANSERAI)` is true, then `city_buildings_has_caravanserai()` must also be true. So you don't need to check if `city_buildings_has_caravanserai()`.

Reply:
yes of course